DOC update - Mfa for the management console (#3149)
wso2/docs-is: en/docs/guides/mfa/mfa-for-management-console.md

# Configuring multi-factor authentication for the management console

 This guide walks you through configuring multi-factor authentication for the Identity Server's Management Console.

 MFA configuration for the management console contains two configuration sections as follows:

1. [Configure the service provider](#configure-the-service-provider)

>     1. [Inbound authentications configurations](#inbound-authentications-configurations)

Reviewer: In this case, the two sub points are not sequential yeah? Then, they can be bullets and not numbers.

>     2. [Local and outbound Authentication configurations](#local-and-outbound-authentication-configurations)

Author: ```suggestion
    - [Local and outbound Authentication configurations](#local-and-outbound-authentication-configurations)
```

2. [Configure the server](#local-configurations-on-deploymenttoml)

## Prerequisites

> You need to [create a service provider]({{base_path}}/guides/applications/register-sp).

Author: ```suggestion
You need to [create a service provider]({{base_path}}/guides/applications/register-sp) to represent the Management Console on the Identity Server. You can use any name as the service provider name.
```

> ## Configure the service provider

Author: Yes, in this case, the management console is considered a normal application on the IS. 

The application specifications like the `Assertion Consumer URLs` is used to identify this application as the management console.

Other than that this has normal SP configurations.

> To configure MFA for the management console, initially:

Author: ```suggestion
To configure MFA for the service provider which represents the management console, initially:
```

2. Click **Add Authentication Step** to add an authentication step.

3. For both authentication steps, select the authenticator from **Local Authenticators** and click **+ Add Authenticator** to add the authenticator.

    !!! info
        - **Use subject identifier from this step** and **Use attributes from this step** will be selected by default. If you wish to set multiple steps, you can have only one step as the subject step and one as the attribute step.
        - Note that you cannot use `jwt-basic` option as an authentication step.

4. Click **Update** to save the configurations.

5. Enable the following options:
    - Use tenant domain in local subject identifier
    - Use user store domain in local subject identifier

    !!! info
        Leave the default configurations as it is.

6. Click **Update** to save the configurations.

## Configure WSO2 IS

To configure the local configuration on the IS:

1. Shut down the Identity server, and open the `deployment.toml` file in the `<IS_HOME>/repository/conf/` directory.

2. Add the following configuration to the `deployment.toml` file:

>     ``` javascript

Reviewer: ```suggestion
    ``` toml
```

> # Configuring multi-factor authentication for the management console

Reviewer: Can't we move this to the Management Console section under Setup?

Author: +1 will move it

> 

Reviewer: need an intro here

## Prerequisites

> You need to [create a service provider](../applications/register-sp.md).

Reviewer: ```suggestion
You need to [create a service provider]({{base_path}}/guides/applications/register-sp).
```

## Configure MFA for the management console

To configure MFA for the management console, initially:

1. On the management console, go to **Main** > **Identity** > **Service Providers** > **List**.

2. Click **Edit** on the service provider you wish to configure.

    Configuration of MFA for the management console contains three configuration sections as follows:

>     1. [Inbound authentications configurations](#inbound-authentications-configurations)

Reviewer: Should these be numbers or bullets? 
Should these be followed in sequence?

Author: 1 and 2 are okay to be interchanged, but 3 should be done as the last step

    2. [Local and outbound Authentication configurations](#local-and-outbound-authentication-configurations)
    3. [Local configurations on `deployment.toml`](#local-configurations-on-deploymenttoml)

### Inbound authentications configurations

To configure inbound authentication for the management console:

> 3. Expand the **Inbound Authentication Configuration > SAML2 Web SSO Configuration** section and click **Configure**.

Reviewer: numbering

4. Enter the following details under **Manual Configuration**:

    | Field name    | Value |
    |---------------|-------|
    | Issuer    | carbonServer  |
    | Assertion Consumer URLs   | https://localhost:9443/acs    |
    | Enable Response Signing   | Selected  |

5. Click **Register** to save the configurations.

### Local and outbound Authentication configurations

To configure local and outbound authentication for the management console:

> 1. Expand the **Local and Outbound Authentication Configuration** section and click **Advanced Configuration**.

Reviewer: ```suggestion
1. Expand **Local and Outbound Authentication Configuration** and click **Advanced Configuration**.
```

2. Click **Add Authentication Step** to add an authentication step.

3. For both authentication steps, select the authenticator from **Local Authenticators** and click **+ Add Authenticator** to add the authenticator.

    !!! info
        - **Use subject identifier from this step** and **Use attributes from this step** will be selected by default. If you wish to set multiple steps, you can have only one step as the subject step and one as the attribute step.
        - Note that you cannot use `jwt-basic` option as an authentication step.

4. Click **Update** to save the configurations.

5. Enable the following options:
    - Use tenant domain in local subject identifier
    - Use user store domain in local subject identifier

    !!! info
        Leave the default configurations as it is.

6. Click **Update** to save the configurations.

> ### Local configurations on `deployment.toml`

Reviewer: This should be a level 2 heading - `Configure WSO2 IS` or `Configure the server`

To configure the local configuration on the IS:

1. Shut down the Identity server, and open the `deployment.toml` file in the `<IS_HOME>/repository/conf/` directory.

2. Add the following configuration to the `deployment.toml` file:

    ``` javascript
    [admin_console.authenticator.saml_sso_authenticator]
    enable=true
    priority="1"
    assertion_consumer_service_url = "https://localhost:9443/acs"
    identity_provider_sso_service_url = "https://localhost:9443/samlsso"
    ```

3. Save the configurations.

## Try it out

1. Start the Identity Server.

2. Access the following URL: `https://localhost:9443/`

3. Enter your user credentials for the first authentication step.

>     ![basic-authentication](../../assets/img/guides/basic-authentication-mfa.png)

Reviewer: let's use base path link
